lib/init.php

```diff
@@ -188,6 +188,25 @@ function register_site_icon_url( $response ) {
 
 add_filter( 'rest_index', 'register_site_icon_url' );
 
+/**
+ * Exposes the site logo to the Gutenberg editor through the WordPress REST
+ * API. This is used for fetching this information when user has no rights
+ * to update settings.
+ *
+ * @since 10.9
+ *
+ * @param WP_REST_Response $response Response data served by the WordPress REST index endpoint.
+ * @return WP_REST_Response
+ */
+function register_site_logo_to_rest_index( $response ) {
+	$data              = $response->data;
+	$data['site_logo'] = get_theme_mod( 'custom_logo' );
+	$response->set_data( $data );
```

Comment:
We can mutate `$response->data` directly.

Let's also add a link if there is a logo set.

```php
$response->add_link(
	'https://api.w.org/featuredmedia',
	rest_url( 'wp/v2/media/' . $logo ),
	array(
		'embeddable' => true,
	)
);
```

Reply:
Done here: https://github.com/WordPress/gutenberg/pull/32919/commits/b0cd792d00478c806d5e68644d66554207830af7